test-it/test.js

```diff
@@ -197,47 +198,71 @@
             }
         }
         var hrstart = process.hrtime();
-        var newResults = tester(testName);
-        var hrend = process.hrtime(hrstart);
-        var msec = hrend[1] / 1000000;
-
-        if (execOnly) {
-            console.log(newResults);
-        } else if (noOldResults) {
-            // Save the generated resutls as the new reference
-            // Add the file name to a list that can be printed at the end.
-        } else {
-            compareResults(testName, oldResults, newResults, msec);
-        }
+        tester(testName, function (newResults) {
+            var hrend = process.hrtime(hrstart);
+            var msec = hrend[1] / 1000000;
+
+            if (execOnly) {
+                console.log(newResults);
+            } else if (noOldResults) {
+                // Save the generated resutls as the new reference
+                // Add the file name to a list that can be printed at the end.
+            } else {
+                compareResults(testName, oldResults, newResults, msec);
+            }
+            testFinished();
+        });
     };
 
     // Process a provided test and return the stdout from the vireo.js runtime
-    var RunVJSTest = function (testName) {
+    var RunVJSTest = function (testName, testFinishedCB) {
         var viaCode;
         var viaPath = 'ViaTests/' + testName;
         vireo.reboot();
         try {
             viaCode = fs.readFileSync(viaPath).toString();
         } catch (e) {
             if (e.code === 'ENOENT') {
-                viaCode = '';
-                throw new Error('No such test ' + testName);
+                viaPath = testName;
+                try {
+                    viaCode = fs.readFileSync(viaPath).toString();
+                } catch (e) {
+                    if (e.code === 'ENOENT') {
+                        viaCode = '';
+                        throw new Error('No such test ' + testName);
+                    }
+                }
             }
         }
+        var testOutput = '';
+        vireo.setPrintFunction(function (text) {
+            testOutput = testOutput + text + '\n';
+        });
 
-        vireo.stdout = '';
         if (viaCode !== '' && vireo.loadVia(viaCode) === 0) {
             // TODO mraj because this is running synchronously we cannot do tests that rely on asynchronous results like http
-            while (vireo.executeSlices(1000000)) {
-                // Nothing here
-            }
+            // TODO spathiwa I think we can now...
```

Comment:
lol almost, we would need to add an XHR polyfill to the global scope in node similar to the following: https://github.com/rajsite/vireo-node-eightball/blob/f7bc85b065b9446b45fd02f38b1554dd79f3949d/server.js#L30